sys/newlib/syscalls.c

```diff
@@ -186,7 +186,7 @@ int _open_r(struct _reent *r, const char *name, int flags, int mode)
  *
  * @return      TODO
  */
-int _read_r(struct _reent *r, int fd, void *buffer, unsigned int count)
+_ssize_t _read_r(struct _reent *r, int fd, void *buffer, size_t count)
```

Comment:
Basically there is no difference. [`sys/_types.h`](https://github.com/eblot/newlib/blob/2a63fa0fd26ffb6603f69d9e369e944fe449c246/newlib/libc/include/sys/_types.h) defines `_ssize_t`, and [`sys/types.h`](https://github.com/eblot/newlib/blob/2a63fa0fd26ffb6603f69d9e369e944fe449c246/newlib/libc/include/sys/types.h) then does `typedef _ssize_t ssize_t`. `sys/types.h` transcludes further headers, and I guess `_ssize_t` is used in prototypes to keep transclusions cycle free.